src/eval.spec.ts

```diff
@@ -0,0 +1,534 @@
+              extraLogging: {},
+            },
+          ],
+          doLog: true,
+        },
+      ],
+      totalShards: 10,
+    };
+
+    const deterministicEvaluator = new Evaluator(
+      new DeterministicSharder({
+        'traffic-alice': 2,
+        'traffic-bob': 3,
+        'traffic-charlie': 4,
+        'traffic-dave': 7,
+        'split-alice': 1,
+        'split-bob': 4,
+        'split-charlie': 8,
+        'split-dave': 1,
+      }),
+    );
+
+    expect(deterministicEvaluator.evaluateFlag(flag, 'alice', {}, false).variation).toEqual(
+      VARIATION_A,
+    );
+    expect(deterministicEvaluator.evaluateFlag(flag, 'bob', {}, false).variation).toEqual(
+      VARIATION_B,
+    );
+    expect(deterministicEvaluator.evaluateFlag(flag, 'charlie', {}, false).variation).toEqual(
+      VARIATION_C,
+    );
+    expect(deterministicEvaluator.evaluateFlag(flag, 'dave', {}, false).variation).toEqual(
+      VARIATION_C,
+    );
+  });
+
+  it('should return none result for evaluation prior to allocation', () => {
+    const now = new Date();
+    const flag: Flag = {
+      key: 'flag',
+      enabled: true,
+      variationType: VariationType.STRING,
+      variations: { a: VARIATION_A },
+      allocations: [
+        {
+          key: 'default',
+          startAt: new Date(now.getFullYear() + 1, 0, 1),
+          endAt: new Date(now.getFullYear() + 1, 1, 1),
+          rules: [],
+          splits: [
+            {
+              variationKey: 'a',
+              shards: [],
+              extraLogging: {},
+            },
+          ],
+          doLog: true,
+        },
+      ],
+      totalShards: 10,
+    };
+
+    const evaluator = new Evaluator(new MD5Sharder());
+    const result = evaluator.evaluateFlag(flag, 'subject_key', {}, false);
+    expect(result.flagKey).toEqual('flag');
+    expect(result.allocationKey).toBeNull();
+    expect(result.variation).toBeNull();
+  });
+
+  it('should return correct variation for evaluation during allocation', () => {
+    const now = new Date();
+    const flag: Flag = {
+      key: 'flag',
+      enabled: true,
+      variationType: VariationType.STRING,
+      variations: { a: VARIATION_A },
+      allocations: [
+        {
+          key: 'default',
+          startAt: new Date(now.getFullYear() - 1, 0, 1),
+          endAt: new Date(now.getFullYear() + 1, 0, 1),
+          rules: [],
+          splits: [
+            {
+              variationKey: 'a',
+              shards: [],
+              extraLogging: {},
+            },
+          ],
+          doLog: true,
+        },
+      ],
+      totalShards: 10,
+    };
+
+    const evaluator = new Evaluator(new MD5Sharder());
+    const result = evaluator.evaluateFlag(flag, 'subject_key', {}, false);
+    expect(result.flagKey).toEqual('flag');
+    expect(result.allocationKey).toEqual('default');
+    expect(result.variation).toEqual(VARIATION_A);
+  });
+
+  it('should evaluate flag after allocation period', () => {
```

Comment:
The test description here IMO should be parallel to the previous ones, e.g. `should return none result for evaluation after allocation end date`. "Evaluate flag" isn't descriptive.